checkEligibility.js

```diff
@@ -36,13 +37,44 @@ function getPatronInfo (patronId) {
   })
 }
 
-function finesBlocksOrExpiration (info) {
+/**
+ *  Given an patron-info hash (i.e. one retrieved via patrons/{patronId}),
+ *  returns a Hash with the following boolean properties:
+ *   - expired: True if card expird
+ *   - blocked: True if card has blocks
+ *   - moneyOwed: True if card has > $15 fines
+ *   - ptypeDisallowsHolds: True if patron ptype disallows holds (e.g. ptype 120, 121)
+ *  Essentially, if Object.keys(hash).map((
+ */
+function identifyPatronIssues (info) {
   info = info.data.entries[0]
-  return {
+  const issues = {
     expired: new Date(info.expirationDate) < new Date(),
     blocked: info.blockInfo.code !== '-', // will want to change this once we have a list of block codes
-    moneyOwed: info.moneyOwed > 15 // may want to change this
+    moneyOwed: info.moneyOwed > 15, // may want to change this
+    ptypeDisallowsHolds: ptypeDisallowsHolds(info.patronType)
   }
+
+  // Set a single property to check for consumers that don't care *what* issues
+  // there are but only that there are issues:
+  issues.hasIssues = Object.keys(issues).filter((name) => issues[name]).length > 0
+
+  return issues
+}
+
+/**
+ *  For a given ptype (e.g. 10, 120), returns `true` if the ptype does not allow
+ *  Sierra holds (which is indicated by the absense of any
+ *  `nypl:deliveryLocationAccess` statement)
+ */
+function ptypeDisallowsHolds (ptype) {
+  const ptypeMapping = nyplCoreObjects('by-patron-type')
+  if (!ptypeMapping) throw new Error('Could not load patron types')
```

Comment:
What do we expect to happen in case this error is thrown? I don't see anything to catch it.

Reply:
Good catch. Added better general error handling. In the case above, it will fall through to [the common catch in index.js](https://github.com/NYPL-discovery/patron-eligibility-service/blob/1fcb83bb21798151f7268fc11d6fc2b209cd03b7/app.js#L19), which will [build a clear error response for the caller](https://github.com/NYPL-discovery/patron-eligibility-service/blob/1fcb83bb21798151f7268fc11d6fc2b209cd03b7/app.js#L26-L32)